Src/AutoFixture/Kernel/FieldSpecification.cs

```diff
@@ -95,7 +95,9 @@ private bool FieldIsCompatibleWithTargetType(object request)
 
         private bool FieldHasTargetName(object request)
         {
-            return ((FieldInfo)request).Name == this.targetName;
+            return ((FieldInfo)request)
+                   .Name
+                   .Equals(this.targetName, StringComparison.OrdinalIgnoreCase);
```

Comment:
I see what you mean. In this particular context, I considered the CS and CI comparisons to be cohesive enough not to break the SRP at the class level.

When it comes to [readability and clarity of intent](http://martinfowler.com/bliki/FlagArgument.html), however, I do agree that flags are easy to misinterpret from the caller's perspective. On top of that, they also bring higher cyclomatic complexity because of the unavoidable selection.

So, yes. I'll go down a different route.